Lib/pathlib.py

```diff
@@ -281,6 +281,13 @@ def _parse_parts(cls, parts):
             path = cls._flavour.join(*parts)
         sep = cls._flavour.sep
         altsep = cls._flavour.altsep
+        if isinstance(path, str):
+            # Force-cast str subclasses to str (issue #21127)
+            path = str(path)
+        else:
+            raise TypeError(
+                "argument should be a str object or an os.PathLike "
+                f"object where __fspath__ returns a str, not {type(path)!r}")
```

Comment:
Sorry for nitpicking here -- I know this is just how it was before -- but it occurs to me that this error message is pretty weird compared to most CPython error messages. Most error messages quote the repr of the _name_ of the class, rather than the repr of the _class itself_. 

```pycon
>>> 1 + 'foo'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for +: 'int' and 'str'
>>> import pathlib
>>> pathlib.Path(b"foo")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\alexw\coding\cpython\Lib\pathlib.py", line 695, in __new__
    self = cls._from_parts(args)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\alexw\coding\cpython\Lib\pathlib.py", line 304, in _from_parts
    drv, root, parts = self._parse_parts(args)
                       ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\alexw\coding\cpython\Lib\pathlib.py", line 288, in _parse_parts
    raise TypeError(
TypeError: argument should be a str object or an os.PathLike object where __fspath__ returns a str, not <class 'bytes'>
```

"A str object" also feels slightly redundant when we can just say "a str" :)

```suggestion
                "argument should be a str or an os.PathLike "
                "object where __fspath__ returns a str, "
                f"not {type(path).__name__!r}")
```

Reply:
All good, let's get it right! I'll apply this locally so I can change the test regex too.